### DISCUSSION TITLE
How to write a query that streams every changes, even when state hasn't changed ?

### DISCUSSION THREAD

**Discussion by brice-noowu** [salesforce/akita#773] [Q&A] [answered]
Thanks to the documentation, I know that 

> The select() method returns an observable that calls distinctUntilChanged() internally, meaning it will only fire when the state changes, i.e., when there is a new reference.

In my usecase, I would like to create a query that streams every changes, even when state hasn't changed.
Is there any ways to override the current `select()` behavior ? Or another way to properly do so with Akita ?

If not, what can I do to make sure I do it right? 

_I'm aware it's a noobish question but i'm new to the RXJS world, and sorry to ask it here, but the gitter is pretty dead (so is the Akita channel on the Angular Discord server)._

**Comment by NetanelBasal** [comment, marked as answer]
What is the use case?

**Comment by brice-noowu** [comment]
To simplify : I have an entity store with a global UI State :
```
ui:{
  scroll:"top" // can be "top" or "bottom"
}
```

I can update it with this method :
```
  updateScrollState(scroll: string) {
    this.update(({ ui }) => ({
      ui: {
        ...ui,
        scroll
      },
    }));
  }
```

In my component I listen for changes on the  `ui.scroll` property thanks to a query declared like this :
`selectScrollState$ = this.select((state) => state.ui.scroll);`

The thing is, I need to do things every time the value is updated, sometimes it can be "top" twice in a row, but for now, the way I do it, nothing is streamed if its the case. (because of distinctUntilChanged())

**Reply by NetanelBasal** [reply]
You can listen to the `ui` object. `this.select((state) => state.ui))`

**Reply by brice-noowu** [reply]
That seems to be working, thank you !
